### ISSUE TITLE
On the issue of model selection

### ISSUE THREAD

**Issue by PIPIKAI** [BIT-DA/CIRL#6]
https://github.com/BIT-DA/CIRL/blob/1b50cd605fa1779b4adc739569371d0f19a6bcce/train.py#L192

**Choose the best model in your training code based on the correctness of the test set rather than the best correctness of the validation set.**

**Comment by SakurajimaMaiii** [comment]
I have the same question.
Previous work, such as [FACT](https://github.com/MediaBrain-SJTU/FACT) uses the **training domain validation**. 
It seems  that this paper chooses the best model on the  (out of distribution) **test set**.

**Comment by JeffLee1874** [comment]
I have the same question.

**Comment by eadstry** [comment]
I have the same question. In Search of Lost Domain Generalization， pointed out that the optimal model on test should not be selected, which leads to overfitting. The appropriate choice is to select the model on the last epoch

**Comment by Linzsd** [comment]
can you reproduce the experiments?

**Comment by undercutspiky** [comment]
Same question.
